### PR TITLE
PDO: Added `PDO::FETCH_CLASS` support on `fetch*()`, `fetchObject()`

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/staabm/phpstan-dba/issues/269

### CODE REVIEW

**Comment by staabm** [comment]
@Seldaek could you double check this one matches your expectations?

File: tests/default/data/pdo-stmt-fetch.php [hunk 1]
```diff
@@ -40,6 +41,12 @@ public function fetchAll(PDO $pdo)
         $all = $stmt->fetchAll(PDO::FETCH_KEY_PAIR);
         assertType('array<int, array<string, int<0, 4294967295>>>', $all);
 
+        $all = $stmt->fetchAll(PDO::FETCH_CLASS, MyRowClass::class);
+        assertType('array<int, staabm\PHPStanDba\Tests\Fixture\MyRowClass>', $all);
+
+        $all = $stmt->fetchAll(PDO::FETCH_CLASS);
+        assertType('array<int, stdClass>', $all);
```
Comment (Seldaek) [inline tests/default/data/pdo-stmt-fetch.php:48]: Looks all good to me, but just to make sure, a simple fetch() here would return stdClass|false, right? Might be worth adding for completeness.

Comment (staabm) [reply tests/default/data/pdo-stmt-fetch.php:48]: fetch() without args depends on a previous setFetchMode(), which needs https://github.com/staabm/phpstan-dba/pull/258 and which in turn waits for the next phpstan release

Comment (Seldaek) [reply tests/default/data/pdo-stmt-fetch.php:48]: Ok I see, great to see it all coming together :)

Comment (staabm) [reply tests/default/data/pdo-stmt-fetch.php:48]: The tricky part was figuring out how this stateful apis can work in phpstan context. 

I don't know any extension which did this before, so this is kind of an achievement

**Pull request merged**